mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -986,6 +995,25 @@ impl TakerSwap {
                 )]))
             },
         };
+        // Validate maker_coin_htlc_pubkey realness
+        match self.taker_coin.validate_other_pubkey(maker_data.maker_coin_htlc_pub()) {
```

Comment:
`maker_coin` should validate `maker_coin_htlc_pub`.
`taker_coin` should validate `taker_coin_htlc_pub`.

Reply:
done